config/initializers/job_configurations.rb

```diff
@@ -1,230 +1,408 @@
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::AgencyInvoiceIaaSupplementReport.new.call },
-)
-
-# Agency Invoice Supplement Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'SP Invoice supplement report by issuer',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::AgencyInvoiceIssuerSupplementReport.new.call },
-)
-
-# Total SP Costs Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'Total SP cost report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::TotalSpCostReport.new.call },
-)
-
-# SP Active Users Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'SP active users report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::SpActiveUsersReport.new.call },
-)
-
-# SP Active Users Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'SP active users over period of peformance report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::SpActiveUsersOverPeriodOfPerformanceReport.new.call },
-)
-
-# Doc auth drop off rates report
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'Doc auth drop off rates report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::DocAuthDropOffRatesReport.new.call },
-)
-
-# IAA Billing Report
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'IAA billing report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::IaaBillingReport.new.call },
-)
-
-# Send Agency User Counts Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'Deleted user accounts report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::DeletedUserAccountsReport.new.call },
-)
-
-# Send GPO Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'GPO report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::GpoReport.new.call },
-)
+if IdentityConfig.store.ruby_workers_enabled
+  # Queue heartbeat job to DelayedJob
+  all_configs[:heartbeat_job] = {
+    good_job: {
+      class: 'HeartbeatJob',
+      cron: cron_5m,
+    },
+  }
+end
 
-# Send Monthly GPO Letter Requests Report to S3
-JobRunner::Runner.add_config JobRunner::JobConfiguration.new(
-  name: 'Monthly GPO letter requests report',
-  interval: 24 * 60 * 60, # 24 hours
-  timeout: 300,
-  callback: -> { Reports::MonthlyGpoLetterRequestsReport.new.call },
-)
+if defined?(Rails::Console)
+  Rails.logger.info 'job_configurations: console detected, skipping schedule'
+elsif IdentityConfig.store.ruby_workers_enabled
+  Rails.application.configure do
+    config.good_job.cron = all_configs.transform_values { |config| config.fetch(:good_job) }
+  end
```

Comment:
the good_job cron scheduler, as written will basically run on all IDP hosts and all worker hosts. I think this okay, because I paired this with `good_job_control_concurrency_with` to make sure only one of each job can be run at one and only 1 can be enqueued at once, for a given day.

There's a possibility one host could queue a day's worth of reports, they could all run, be emptied out, and then a second host could queue them all again and they'd run.

However, since the cron schedule means they're aiming for the same minute, I think we'll be okay here.

And at the end of the day, if the reports are so quick that they can run within the a minute's worth of clock drift, that's probably easier to leave it than to try to do leader election of the cron scheduler